<a href="https://colab.research.google.com/github/HarshaVardhaniMundeshwar/-Emerging-AI-Technologies/blob/main/Assignment_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Group Assignment 1 – Machine Learning Problem Proposal**




# By Group 3
## Harsha Vardhani Mundeshwar
## Jay Bankim Patel
## Mahendra Sharma
## Pragna Palisetty Vinayaka


# Industry Chosen: Healthcare
# Scenario: Hospital Readmission Risk Prediction System

## **1. Detailed Problem Statement**

**The Issue / Context**

In modern healthcare systems, the 30-day hospital readmission rate is a major performance indicator. A readmission occurs when a patient is discharged but must return to the hospital within 30 days due to complications or worsening conditions.

**Actual Problem / Situation**

Most hospitals still use generalized discharge procedures based on clinical guidelines rather than personalized risk assessment. Physicians lack a data-driven tool to determine which patients are at high risk of returning. As a result:

- Preventable relapses occur
- Emergency departments become overcrowded
- Hospital costs increase
- Patient health outcomes worsen

**How and Why It Happens**

Hospital readmission is influenced by multiple interacting factors, including age, chronic diseases, medication history, and past hospital visits. These relationships are too complex for manual analysis during discharge decisions. Traditional rule-based systems fail to capture subtle nonlinear patterns hidden in large clinical datasets.

Machine learning can analyze thousands of past patient records and detect these hidden patterns, making it suitable for predicting readmission risk.

**The Relevance**

This problem has serious consequences:

Hospitals face financial penalties under programs like the Hospital Readmissions Reduction Program (HRRP)

Readmissions cost healthcare systems bi
llions annually

Patients face increased complications and emotional stress

Hospital resources are strained

Even a small reduction (5–10%) in readmissions can significantly improve healthcare efficiency and patient safety.

**Purpose, Vision, and Scope**

Vision:
To provide doctors with an automated Readmission Risk Score at the time of discharge to support clinical decision-making.

Purpose:
Enable hospitals to identify high-risk patients early and apply preventive interventions such as follow-up calls, medication reviews, or home monitoring.

Scope:

Use structured electronic health record (EHR) data

Focus only on 30-day readmission prediction

Develop a binary classification model

Support doctors rather than replace clinical judgment


## **2. Hypothesis & Proposed ML Solution**

**Hypothesis**

By implementing an ensemble machine learning model such as Random Forest or XGBoost, we can identify high-risk patients with a Recall greater than 0.80, outperforming traditional clinical scoring methods and enabling targeted interventions that reduce readmission rates.

**Proposed ML Solution**

| Component            | Description                                                                     |
| -------------------- | ------------------------------------------------------------------------------- |
| **ML Type**          | Supervised Learning – Binary Classification                                     |
| **Input Features**   | Age, diagnosis history, number of prior admissions, lab results, length of stay |
| **Output**           | 1 = Readmitted, 0 = Not Readmitted                                              |
| **Candidate Models** | Logistic Regression (baseline), Random Forest, XGBoost                          |
| **Goal**             | Predict probability of readmission before discharge                             |



## **3. Peer-Reviewed Research Review & Comparison**

| Research Paper             | Approach                                     | Key Findings                                                              | How It Supports Our Work                                     |
| -------------------------- | -------------------------------------------- | ------------------------------------------------------------------------- | ------------------------------------------------------------ |
| **Rajkomar et al. (2018)** | Deep Learning on EHR                         | ML models significantly outperformed traditional clinical scoring systems | Confirms that EHR data is powerful for outcome prediction    |
| **Futoma et al. (2015)**   | Random Forest & Gradient Boosting            | Ensemble models achieved better stability and accuracy                    | Supports our choice of Random Forest and boosting algorithms |
| **Choi et al. (2016)**     | Recurrent Neural Networks on patient history | Captured temporal patterns in medical records for prediction              | Highlights importance of using longitudinal health data      |


## **4. Evaluation Strategy**

To ensure reliable and clinically meaningful results, we will evaluate the model using:

- Recall → Minimizes false negatives (missing a high-risk patient)
- ROC-AUC → Measures overall discrimination ability
- F1-Score → Balances precision and recall

Healthcare prediction prioritizes Recall because failing to detect a high-risk patient can have severe consequences.

# **5. References**

Choi, E., Schuetz, A., Stewart, W. F., & Sun, J. (2016). Using recurrent neural network models for early detection of heart failure onset. Journal of the American Medical Informatics Association, 24(2), 361–370. https://doi.org/10.1093/jamia/ocw112

Futoma, J., Morris, J., & Lucas, J. (2015). A comparison of models for predicting early hospital readmissions. Journal of Biomedical Informatics, 56, 229–238. https://doi.org/10.1016/j.jbi.2015.05.016

Rajkomar, A., Oren, E., Chen, K., Dai, A. M., Hajaj, N., Hardt, M., & Dean, J. (2018). Scalable and accurate deep learning with electronic health records. npj Digital Medicine, 1(18). https://doi.org/10.1038/s41746-018-0029-1

# **ABOVE & BEYOND SECTION**

**Preliminary Machine Learning Prototype**

To demonstrate the feasibility of the proposed readmission prediction system, we developed a small prototype using a public healthcare dataset. This prototype shows how a machine learning classification model can be trained and evaluated to predict patient readmission risk.


## 1. Import Libraries

In [1]:
# Import necessary libraries for machine learning
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import classification_report, roc_auc_score


## 2. Load Sample Dataset

In [4]:
from sklearn.datasets import load_breast_cancer
import pandas as pd

# Load dataset
cancer = load_breast_cancer()

# Convert to DataFrame
data = pd.DataFrame(data=cancer.data, columns=cancer.feature_names)
data['target'] = cancer.target  # 1 = malignant, 0 = benign

data.head()


,mean radius,mean texture,mean perimeter,mean area,mean smoothness,mean compactness,mean concavity,mean concave points,mean symmetry,mean fractal dimension,...,worst texture,worst perimeter,worst area,worst smoothness,worst compactness,worst concavity,worst concave points,worst symmetry,worst fractal dimension,target
0,17.99,10.38,122.80,1001.0,0.11840,0.27760,0.3001,0.14710,0.2419,0.07871,...,17.33,184.60,2019.0,0.1622,0.6656,0.7119,0.2654,0.4601,0.11890,0
1,20.57,17.77,132.90,1326.0,0.08474,0.07864,0.0869,0.07017,0.1812,0.05667,...,23.41,158.80,1956.0,0.1238,0.1866,0.2416,0.1860,0.2750,0.08902,0
2,19.69,21.25,130.00,1203.0,0.10960,0.15990,0.1974,0.12790,0.2069,0.05999,...,25.53,152.50,1709.0,0.1444,0.4245,0.4504,0.2430,0.3613,0.08758,0
3,11.42,20.38,77.58,386.1,0.14250,0.28390,0.2414,0.10520,0.2597,0.09744,...,26.50,98.87,567.7,0.2098,0.8663,0.6869,0.2575,0.6638,0.17300,0
4,20.29,14.34,135.10,1297.0,0.10030,0.13280,0.1980,0.10430,0.1809,0.05883,...,16.67,152.20,1575.0,0.1374,0.2050,0.4000,0.1625,0.2364,0.07678,0


## Model Training

In [6]:

# Split into features and target
X = data.drop('target', axis=1)
y = data['target']

# Train-test split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Train a machine learning model
model = RandomForestClassifier(random_state=42)
model.fit(X_train, y_train)

# Evaluate the model
y_pred = model.predict(X_test)
print(classification_report(y_test, y_pred))


              precision    recall  f1-score   support

           0       0.98      0.93      0.95        43
           1       0.96      0.99      0.97        71

    accuracy                           0.96       114
   macro avg       0.97      0.96      0.96       114
weighted avg       0.97      0.96      0.96       114



The Random Forest model achieved an accuracy of 96% on the test dataset. The recall score for the positive class is 0.99, which indicates that the model is highly effective at identifying true positive cases. This is especially important in healthcare prediction problems where missing a high-risk case can have serious consequences.

Although this prototype uses a sample dataset, the results demonstrate that machine learning classification models can achieve strong predictive performance. This supports the feasibility of our proposed hospital readmission risk prediction system, which will be implemented using real hospital datasets in Assignment 2.